packages/hint-doctype/src/hint.ts

```diff
@@ -0,0 +1,150 @@
+
+            lines.forEach((line: string, i: number): void => {
+                if (found) {
+                    return;
+                }
+
+                const matched = doctypeRegExpFlexible.exec(line);
+
+                if (matched){
+                    found = true;
+                    location.line = i;
+                    location.column = matched.index;
+                }
+            });
+
+            return location;
+        };
+
+        const getMatchInformation = (text: string): MatchInformation => {
+            return {
+                location: getCurrentDoctypeProblemLocation(text),
+                matches: text.match(doctypeRegExpFlexible)
+            };
+        };
+
+        const checkNoDoctypeInContent = async (matchInfo: MatchInformation, resource: string): Promise<boolean> => {
+            if (matchInfo.matches && matchInfo.matches.length > 0) {
+                return true;
+            }
+
+            await report(resource, 'The resource does not contain a valid DOCTYPE.');
+
+            return false;
+        };
+
+        const checkNoDoctypeInCorrectLine = async (matchInfo: MatchInformation, resource: string): Promise<void> => {
+            if (matchInfo.location.line === correctLine) {
+                return;
+            }
+
+            await report(resource, 'DOCTYPE is not in the first line.', matchInfo.location);
+        };
+
+        const checkDoctypeHasMoreThanValidInfo = async (matchInfo: MatchInformation, resource: string): Promise<void> => {
+            if (!matchInfo.matches || matchInfo.matches.length < 1) {
+                return;
+            }
+
+            const matchedDoctype = matchInfo.matches[0];
+            const strictMatch = matchedDoctype.match(doctypeRegExp);
+
+            if (!strictMatch || strictMatch.length < 1) {
+                return;
+            }
+
+            if (strictMatch[0] === matchedDoctype) {
+                return;
+            }
+
+            await report(resource, 'There is additional information on the line with the DOCTYPE.');
+        };
+
+        const checkDoctypeIsDuplicated = async (matchInfo: MatchInformation, resource: string): Promise<void> => {
+            if (!matchInfo.matches || matchInfo.matches.length < 2) {
+                return;
+            }
+
+            await report(resource, 'There is more than one DOCTYPE in the document.', matchInfo.location);
+
+        };
+
+        const onFetchEndHTML = async (fetchEnd: FetchEnd): Promise<void> => {
+            const { resource, response } = fetchEnd;
+
+            if (!response || !response.body || !response.body.content) {
+                await context.report(resource, null, 'Resource has no content.', undefined);
+
+                return;
+            }
+
+            const { body } = response;
+            const { content } = body;
+
+            const globalMatch = getMatchInformation(content);
+
+            if (!checkNoDoctypeInContent(globalMatch, resource)) {
```

Comment:
I think you need to `await` here to get the right result. Otherwise I think you are just casting a Promise to a boolean.